server/jest-setup.js

```diff
@@ -1,27 +1,10 @@
 // mock* prefix allows ref to out of scope var inside mock
-let mockTestDbName
-
-jest.mock('pubsweet-server/src/db', () => {
-  const pg = require('pg')
-  const logger = require('@pubsweet/logger')
-  mockTestDbName = `test_${Date.now()}_${Math.floor(Math.random() * 9999999)}`
-  const pool = new pg.Pool({ database: mockTestDbName })
-  pool.on('error', err => {
-    if (err.message !== 'terminating connection due to administrator command') {
-      logger.error(err)
-    }
-  })
-  pool.testDbName = mockTestDbName
-
-  return pool
-})
```

Comment:
Removing this stuff means we can no longer parallelise test runs (need to run jest with `--runInBand` to avoid errors on multi-core machines). But we have so few tests that for the time being it's only a couple of seconds faster when running in parallel. The gain in simplicity is probably worth the loss in performance.

Reply:
It wasn't the simplicity ... it was the fact that `jest.mock` doesn't work. There are three places `db.js` is used in the project and the jest mock would only work for one of them. I did try to explicitly mock out the paths but to no avail. Do you know how to ensure that jest mocks everything under `node_modules` ? @tamlyn

Reply:
No, I just had a look too and gave up. It's a hacky solution which is good when it works but hard to debug when it doesn't. That's what I means about simplicity. At least it's easier to reason about what's going on when you have a single database.

Reply:
Agreed,  just a shame it doesn't work. I just saw your commit (e615fc1) to put in the "runInBand" flag, looks good. I was wondering where is the best place to put in the tests for `selectById` (and status) to test the queries.... At the `data-access` level it gets tested to make sure it returns the correct object id ... but it doesn't check exactly what is returned. I'm thinking this should go in the `resolve.test` ...? ... what's the thinking @damnhipster  @tamlyn ?

Reply:
With Hem's help I realised that the root of the problem here was that `pubsweet-server` was present multiple times in the dependency tree. This is a problem because each one gets its own database pool and it introduces subtle problems as seen here. Ensuring that all pubsweet dependencies are always upgraded together avoids this.

I've re-enabled parallel test execution for now. It will become more valuable as the number of tests increases.

Reply:
Oh wow! That is fantastic! So how was the dependency tree fixed ... we'll talk tomorrow - great work again :+1: